add context manager support for ImcHandle

DanielCorral opened the pull request:
- moves `ImcHandle.login` defaults to `ImcHandle.__init__`
    this is so `ImcHandle._login` and `ImcHandle.__enter__`
    use the same defaults.

Commit:
add context manager support for ImcHandle

moves ImcHandle.login defaults to __init__
    this is so ImcHandle._login and ImcHandle.__enter__
    use the same defaults.

waghswapnil commented:
@DanielCorral : Thanks for this very useful feature !
Can you please also put in a small unit test for the same in `tests/session/test_imcsession.py` ?
I will also update the documentation with this new usage.
And lastly, thanks for cleaning up some of the typos.

waghswapnil commented:
@DanielCorral : If we skip validation, the `model` and `platform` will not be populated in the handle.
The schema for C3260/S3260 platforms differs from the regular C220/240/460 etc. platforms and these attributes help us in identifying the right set of Managed Objects and properties relevant to that platform.
If these attributes in the handle are not populated, it will create issues when resolving properties specific to a platform.
Can we remove the `skip validation` logic from the changes and resubmit the PR, so that the rest of the changes can be merged ?

DanielCorral commented:
@waghswapnil I was unaware of this, good that you caught this ahead of time. Anyway I have removed the validation option and added a simple unit test.
